### ISSUE TITLE
Labels from env is not being expanded in config

### ISSUE THREAD

**Issue by antonzhyliuk** [aws-observability/aws-otel-collector#748]
**Describe the bug**
config for prometheus containing env vars which is not being expanded.
```
receivers:
  prometheus:
    config:
      global:
        scrape_interval: 1m
        scrape_timeout: 10s

      scrape_configs:
        - job_name: jmx
          static_configs:
            - targets: ["service:8080"]
              labels:
                service: toaster
                runtime_environment: "${RUNTIME_ENVIRONMENT}"
                deployment: "${DEPLOYMENT}"
```

when I read metrics from prometheus I could find metrics with label runtime_environment being set as plain string `"${RUNTIME_ENVIRONMENT}"`

**What did you expect to see?**
I expect to see value of env var being set to the label

**Environment**
I see the problem in the latest version: 0.14.0

**Additional context**
I think it worked well on the previous versions. I'll try to find which one is working

**Comment by anuraaga** [comment]
@antonzhyliuk Do you know if it works without the quotes? e.g.

`runtime_environment: $RUNTIME_ENVIRONMENT`

I haven't used env vars for this setting in particular before but have for others using that format

**Comment by antonzhyliuk** [comment]
@anuraaga thanks for reply, I tried without any success

**Comment by ericksoen** [comment]
I experienced similar issues when converting from the `otel/opentelemetry-collector` image to `amazon/aws-otel-collector:v0.14.1`. I was able to use [environment variable expansion](https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables) in the upstream `otel/opentelemetry-collector` without issue.

**Comment by antonzhyliuk** [comment]
@ericksoen thanks for reply and picking the latest stable :) Currently I've built a workaround and not rely on env-vars anymore in my configs, but it would be nice to have this functionality since open telemetry project configuration supports it:
https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables

**Comment by jefchien** [comment]
@antonzhyliuk Are you also using the environment variable `AOT_CONFIG_CONTENT` to set your config?

**Comment by jefchien** [comment]
@ericksoen @antonzhyliuk @mystical44 We're planning to release patches for v0.14.x and v0.15.x that include the fix and will let you know when those are available.

**Comment by jefchien** [comment]
@ericksoen @antonzhyliuk We've released [v0.14.2](https://github.com/aws-observability/aws-otel-collector/releases/tag/v0.14.2) and [v0.15.1](https://github.com/aws-observability/aws-otel-collector/releases/tag/v0.15.1) (latest). Please use one of these releases.
